### PR TITLE
[rllib] Rename sample_batch_size => rollout_fragment_length

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

A common source of confusion is the difference between train_batch_size and sample_batch_size, since both refer to the size of an object of type `SampleBatch`.

It is difficult to be rid of the sample batch concept since a distributed system necessarily batches at multiple levels for efficiency. However, we can at least rename it to clear things up a little.

The new dogma is: `rollout workers produce batches of experiences `rollout_length` steps in size`.

### CODE REVIEW

File: doc/source/rllib-algorithms.rst [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
 
+.. tip::
+
+    Check out the `environments <rllib-env.html>`__ page to learn more about different environment types.
+
+Feature Compatibility Matrix
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+=============  =======================  ==================  ===========  ===========================
+Algorithm      Discrete Actions         Continuous          Multi-Agent  Model Support
```
Comment (sven1977) [inline doc/source/rllib-algorithms.rst]: "Continuous" -> "Continuous Actions"

File: rllib/evaluation/rollout_worker.py [hunk 2]
```diff
@@ -435,7 +435,7 @@ def make_env(vector_index):
                 self.preprocessors,
                 self.filters,
                 clip_rewards,
-                unroll_length,
+                rollout_fragment_length,
```
Comment (sven1977) [inline rllib/evaluation/rollout_worker.py]: Nice.

**Review by sven1977 — APPROVED** [review]
Cool, thanks for cleaning this up!

File: doc/source/rllib-algorithms.rst (hunk 1 above)
Comment (ericl) [reply doc/source/rllib-algorithms.rst]: The table gets too wide then unfortunately

**Pull request merged**